packages/utils/src/instrument.ts

```diff
@@ -259,6 +259,9 @@ function instrumentXHR(): void {
           };
         });
       } else {
+        logger.warn(
+          "XHR instance has no `onreadystatechange` handler attached at the time of calling it's `send` method. Consider changing the order, as it may affect breadcrumbs/tracing capturing. Or add an empty handler to remove this warning.",
+        );
```

Comment:
Hmm 🤔 
Not sure if this is what we want, this would fire also if someone just wants to do a simple XHR request, especially this part 
> Or add an empty handler to remove this warning.

Can we do the following, when we patch `send` we toggle a flag, if our `onreadystatechange` implementation is called this flag is toggled and we never check for it again.
If this flag is never the desired value (when someone calls send again) we know for sure our `onreadystatechange` was never called. Then we can even emit an error with a more "call to action" message.

```
let flag: boolean?;
// undefined until the first `send` call
// true after first send
// we set it to false in our invocation of `onreadystatechange`
// if flag === true in next send call we bust
```